librz/core/cmd/cmd_print.c

```diff
@@ -1716,48 +1716,83 @@ static void pr_bb(RzCore *core, RzAnalysisFunction *fcn, RzAnalysisBlock *b, boo
 	}
 }
 
-static void disasm_until_ret(RzCore *core, ut64 addr, int limit, RzOutputMode mode) {
-	int p = 0;
+static void handle_default_disasm_print_mode(const RzCore *core, const ut64 addr, const bool ret_val,
+	const char *m_intr, RzStrBuf *buf) {
 	const bool show_color = rz_config_get_i(core->config, "scr.color");
+
+	if (show_color) {
+		const char *offsetColor = rz_cons_singleton()->context->pal.offset; // TODO etooslow. must cache
```

Comment:
What is the todo about? Maybe better add it to an issues which things you still need to improve in general.